lib/streamlit/components/v1/custom_component.py

```diff
@@ -101,6 +102,7 @@ def create_instance(
 And if you're using Streamlit Cloud, add "pyarrow" to your requirements.txt."""
             )
 
+        check_widget_usage()
```

Comment:
Does this mean custom components cannot be used  in cached functions anymore? Is this a new, potentially breaking behavior?

Reply:
The `component_instance` was also part of the previous list of widgets: 

https://github.com/streamlit/streamlit/pull/8503/files#diff-0c23722034bf09ab061b57063170e8f84a747adb6fee9f2e9bdc80cf724d80d8L21

So, I also believe it isn't currently possible to use custom components inside cached functions.

Reply:
Ahh ok, thanks for the pointer!